opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1212,6 +1213,328 @@ Dune::cpgrid::Intersection CpGrid::getParentIntersectionFromLgrBoundaryFace(cons
+            const auto& bornLevel_bornIdx =  (*current_data_)[level]->corner_history_[point.index()];
+            if (bornLevel_bornIdx[0] != -1)  { // Corner in the refined grid coincides with a corner from level 0.
+                // Therefore, search and assign the global id of the previous existing equivalent corner.
+                const auto& equivPoint = cpgrid::Entity<3>(*( (*current_data_)[bornLevel_bornIdx[0]]), bornLevel_bornIdx[1], true);
+                localToGlobal_points_per_level[level-1][point.index()] = current_data_->front()->global_id_set_->id( equivPoint );
+            }
+            else {
+                // Assign CANDIDATE global id to (all partition type) points that do not coincide with
+                // any corners from level zero.
+                // TO DO after invoking this method: make a final decision on a unique id for points of refined level grids,
+                // via a communication step.
+                localToGlobal_points_per_level[level-1][point.index()] = min_globalId_point_in_proc;
+                ++min_globalId_point_in_proc;
+            }
+        }
+    }
+}
+
+void CpGrid::selectWinnerPointIds([[maybe_unused]] std::vector<std::vector<int>>&  localToGlobal_points_per_level,
+                                  [[maybe_unused]] const std::vector<std::tuple<int,std::vector<int>>>& parent_to_children,
+                                  [[maybe_unused]] const std::vector<std::array<int,3>>& cells_per_dim_vec) const
+{
+#if HAVE_MPI
+    // To store cell_to_point_ information of all refined level grids.
+    std::vector<std::vector<std::array<int,8>>> level_cell_to_point(cells_per_dim_vec.size());
+    // To decide which "candidate" point global id wins, the rank is stored. The smallest ranks wins,
+    // i.e., the other non-selected candidates get rewritten with the values from the smallest (winner) rank.
+    std::vector<std::vector<int>> level_winning_ranks(cells_per_dim_vec.size());
+
+    for (std::size_t level = 1; level < cells_per_dim_vec.size()+1; ++level) {
+
+        level_cell_to_point[level -1] = currentData()[level]->cell_to_point_;
+        // Set std::numeric_limits<int>::max() to make sure that, during communication, the rank of the interior cell
+        // wins (int between 0 and comm().size()).
+        level_winning_ranks[level-1].resize(currentData()[level]->size(3), std::numeric_limits<int>::max());
+
+        for (const auto& element : elements(levelGridView(level))) {
+            // For interior cells, rewrite the rank value - later used in "point global id competition".
+            if (element.partitionType() == InteriorEntity) {
+                for (const auto& corner : currentData()[level]->cell_to_point_[element.index()]){
+                    int rank = comm().rank();
+                    level_winning_ranks[level -1][corner] = rank;
+                }
+            }
+        }
+    }
```

Comment:
I see your point, but the Handle does not see any grid, so changing the for-loop to elemIdx with size of level_cell_to_point_  would be okay, but then to detect if the each element is interior we need an Entity<0>. To create an Entity, we need each refined level grid. 
It can be done, but implies passing to the constructor of the handle all refined level grids.